parcels/include/parcels.h

```diff
@@ -73,6 +73,65 @@ static inline ErrorCode spatial_interpolation_bilinear(double xsi, double eta, f
   return SUCCESS;
 }
 
+/* Bilinear interpolation routine for 2D grid for tracers with inverse distance weighting near land*/
+static inline ErrorCode spatial_interpolation_bilinear_invdist_land(double xsi, double eta, float data[2][2], float *value)
+{
+  int i, j, k, l, nb_land = 0, land[2][2] = {{0}};
+  float weight[2][2] = {{0.}}, w_sum = 0.;
+  // count the number of surrounding land points (assume land is where the value is close to zero)
+  for (i = 0; i < 2; i++) {
+    for (j = 0; j < 2; j++) {
+      if (is_zero_flt(data[i][j])) {
+	    land[i][j] = 1;
+	    nb_land++;
+      }
+      else {
+	    // record the coordinates of the last non-land point
+	    // (for the case where this is the only location with valid data)
+	    k = i;
+	    l = j;
+      }
+    }
+  }
+  switch (nb_land) {
+  case 0:  // no land, use usual routine
+    return spatial_interpolation_bilinear(xsi, eta, data, value);
+  case 3:  // single non-land point
+    *value = data[k][l];
+    return SUCCESS;
+  case 4:  // only land
+    return 0.;
```

Comment:
This `return 0` is not the cleanest way to do this. In parcels, the way you would normally handle these `ERROR_INTERPOLATION` is with a recovery kernel. So I would suggest to keep `return ERROR_INTERPOLATION`

(by the way, also note that since these errorcodes are simple enums, `return 0` is the same as `return SUCCESS`; see https://github.com/OceanParcels/parcels/blob/master/parcels/include/index_search.h#L19-L22)